functions/src/modules/SQR.functions.ts

```diff
@@ -214,65 +215,71 @@ export const processSubmissions = functions.database.ref('/webforms/sqr/{submiss
 //          on FB (Storage Triggered)
 //
 /////////////////////////////////////////////////
+const path = require('path');
+const os = require('os');
+const fs = require('fs');
 
-export const updateEmailTemplates = functions.storage.object()
-.onFinalize(async object => {
-
-    const path = require('path');
-    const os = require('os');
-    const fs = require('fs');
-
-    const filePath = object.name;
-    // slice(0, -5) to get rid of the trailing '.html' extension
-    const fileName = filePath.split('/')[2].slice(0, -5);
-    const tempLocalFile = path.join(os.tmpdir(), fileName);
-    
+const updateEmailTemplates = async (filePath) => {
+    // slice(0, -5) to get rid of the trailing '.mustache' extension
+    const fileName = filePath.split('/')[2].slice(0, -9);    
     
     if (!filePath.startsWith('email/templates')) 
-        return 1;
-
-    let emailRef = bucket.file(object.name);
-    await emailRef.download({ destination: tempLocalFile });
-    
-    // Use the following to upate the template on SendInBlue
-    let htmlContent = fs.readFileSync(tempLocalFile, 'utf-8');
+        return;
 
-    
-    //////////////////////////////////////////
     // 1. Get the template ID (sendInBlue ID)
-    //////////////////////////////////////////
-
     let templateNode = await db.ref('/email/templates').orderByKey()
-    .equalTo(fileName).once('value');
-
+                                .equalTo(fileName).once('value');
     let template = templateNode.val();
 
     if (template.exists()) {
-        await helpers.updateTemplate(template.id, htmlContent);
-        /////////////////////////////////////////////////////////////////
+        
+        const tempLocalFile = path.join(os.tmpdir(), fileName);
+        let emailRef = bucket.file(filePath);    
+        await emailRef.download({ destination: tempLocalFile });
+
+        // Use the following to upate the template on SendInBlue
+        let htmlContent = fs.readFileSync(tempLocalFile, 'utf-8');
+        let { newTemplate, sender, bcc, subject, sample } = template;
+        let id;
+
+        if (newTemplate) {
+            id = await emailHelper.createTemplate(fileName, htmlContent, subject);
+            await db.ref(`/email/templates/${fileName}`).push({
```

Comment:
You should not push into this path. The template metadata should be created manually.